pkg/api/job_runs.go

```diff
@@ -116,6 +121,99 @@ func FetchJobRun(dbc *db.DB, jobRunID int64, logger *log.Entry) (*models.ProwJob
+// compareRelease and the prowJob variants, starting with the full name.  When no match is found it will iterate while
+// removing the leading 'string-'
+// and try to find a match until successful or no matches are found.
+//
+// The use case is for pull request jobs that we want to find a matching periodic that is running the
+// same root job.  We use the periodic as the 'standard' to compare test rates.
+// e.g.
+// pull-ci-openshift-origin-master- e2e-vsphere-ovn-etcd-scaling
+// periodic-ci-openshift-release-master-nightly-4.14- e2e-vsphere-ovn-etcd-scaling
+// our common root is e2e-vsphere-ovn-etcd-scaling and our compareRelease is 4.14
+// if we don't have enough data from the current compareRelease we fall back to include the previous release as well
+func findReleaseMatchJobNames(dbc *db.DB, jobRun *models.ProwJobRun, compareRelease string, logger *log.Entry) ([]string, int, error) {
+	segments := strings.Split(jobRun.ProwJob.Name, "-")
+
+	// if we don't find enough jobs to match against we can try the prior release
+	// and see if it has enough, think about cutover to a new release, etc.
+
+	for i := 0; i < len(segments); i++ {
+
+		// pull-ci-openshift-origin-master-e2e-vsphere-ovn-etcd-scaling
+		// ci-openshift-origin-master-e2e-vsphere-ovn-etcd-scaling
+		// openshift-origin-master-e2e-vsphere-ovn-etcd-scaling
+		// origin-master-e2e-vsphere-ovn-etcd-scaling
+		// master-e2e-vsphere-ovn-etcd-scaling
+		// e2e-vsphere-ovn-etcd-scaling
+		// matches periodic-ci-openshift-release-master-nightly-4.14-e2e-vsphere-ovn-etcd-scaling
+		// when we specify the 4.14 release
+		name := joinSegments(segments, i, "-")
+
+		if len(name) > 0 {
+			jobs, err := query.ProwJobSimilarName(dbc, name, compareRelease)
+
+			if err != nil {
+				logger.WithError(err).Errorf("Failed to find similar name for release: %s, root: %s", compareRelease, name)
+			}
+
+			if len(jobs) > 0 {
+				logger.Infof("Found %d matches with: %s", len(jobs), name)
+
+				// the first hit we get
+				// compare the variants
+				// for the matches
+				// query the run count for each id
+				// and total it up
+
+				allJobNames := make([]string, 0)
+				totalJobRunsCount := 0
+				hasNeverStableJob := false
+				variants := jobRun.ProwJob.Variants
+				gosort.Strings(variants)
+				for _, job := range jobs {
+					// this is a weird way to get the variant we want, but it allows re-use
+					// of the existing code.
+					// how do we handle never-stable
+					if len(job.Variants) == 1 && job.Variants[0] == testidentification.NeverStable {
+						hasNeverStableJob = true
+					} else {
+						gosort.Strings(job.Variants)
+						if stringSlicesEqual(variants, job.Variants) {
+
+							jobIds, err := query.ProwJobRunIds(dbc, job.ID)
+
+							if err != nil {
+								logger.WithError(err).Errorf("Failed to query job run ids for %d", job.ID)
+								continue
+							}
+
+							totalJobRunsCount += len(jobIds)
+							allJobNames = append(allJobNames, "'"+job.Name+"'")
```

Comment:
We should log when we find a matching job name, perhaps at debug level, either here or when we get the final list back out of the func.

Reply:
Disregard I saw later it was in the result message.